### PR TITLE
More parser tests

### PR DESCRIPTION
**Pull request by atodorov**
This needs to be rebased once #15 has been merged. 

@clumens @dashea can you look at the test for `parseOneTag`. The bytestream is taken from a vlc src rpm I had lying around. From what I can tell we first parse the signature section but it looks like my data contains tags which have nothing to do with the signature. This is something I don't quite understand. 
I've tried to comment the relevant sections in the code to make sense of the binary data.

### CODE REVIEW

**Comment by clumens** [comment]
@atodorov - Sorry, I'm having a bit of a problem following where you're running into problems.  Can you link to the lines in question?  My first thought was perhaps you haven't taken the RPM lead into account, but it looks like you have.  My second thought is perhaps you haven't taken into account that the signature section gets padded.

File: tests/RPM/ParseSpec.hs [hunk 1]
```diff
@@ -125,3 +127,87 @@ spec = describe "RPM.Parse" $ do
...
+      let stream = BS.pack [
+            0x8E, 0xAD, 0xE8, -- section header signature
+            1, -- sectionVersion
+            0, 0, 0, 0, -- 4 reserved bytes
+            0, 0, 0, 7, -- sectionCount 4 bytes
+            0, 0, 0, 0xE4, -- sectionSize 4 bytes
+
+            -- tags defined in this section, sectionCount * 16 bytes
+            -- 62 7 212 16 == HeaderSignatures (Null), ty(7) /= 0, returns Nothing
+            0x00, 0x00, 0x00, 0x3e, 0x00, 0x00, 0x00, 0x07, 0x00, 0x00, 0x00, 0xd4, 0x00, 0x00, 0x00, 0x10,
```
Comment (atodorov) [inline tests/RPM/ParseSpec.hs]: @clumens I was referring to this part of the header but then I figured it out. The ty values here do not match what the code expects so mkTag returns Nothing. I guess this is from an older version of RPM.

File: tests/RPM/Parse_parseRPMSpec.hs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                                                                                                       0x41,
+            0x20, 0x66, 0x72, 0x65, 0x65, 0x20, 0x61, 0x6e,  0x64, 0x20, 0x63, 0x72, 0x6f, 0x73, 0x73, 0x2d,
+            0x70, 0x6c, 0x61, 0x74, 0x66, 0x6f, 0x72, 0x6d,  0x20, 0x6d, 0x65, 0x64, 0x69, 0x61, 0x20, 0x70,
+            0x6c, 0x61, 0x79, 0x65, 0x72, 0x00, -- RPM payload -- MODIFIED FOR BREVITY
+                                                0x54, 0x45,  0x53, 0x54, 0x2D, 0x54, 0x45, 0x53, 0x54, 0x21]
+      -- parsing succeeds
+      parseRPM `shouldSucceedOn` stream
+      -- no unconsumed input
+-- TODO: I am not sure why this fails ATM, it says parser didn't leave anything behind
+--      stream ~?> parseRPM `leavesUnconsumed` BS.pack []
```
Comment (atodorov) [inline tests/RPM/Parse_parseRPMSpec.hs]: @clumens however now I have a new problem. This line fails with:
```
  src/Test/Hspec/Attoparsec.hs:129: 
  1) RPM.Parse_parseRPM.RPM.Parse.parseRPM succeeds with valid data
         expected the parser to leave the following unconsumed: ""
         but got no unconsumed input
```
From what I can tell the parser leaves `Nothing` unconsumed hence the failure. What I'm still trying to figure out is why I don't get the same failure at all the other places I use a similar assertion.

Comment (clumens) [inline tests/RPM/Parse_parseRPMSpec.hs]: I think the difference is that parseRPM ends with a call to takeByteString.  The docs for that function says it grabs all remaining input and puts that into a single string.  Nowhere else do we call that function.  Perhaps its behavior is what leads to the failure you are seeing.

Comment (atodorov) [inline tests/RPM/Parse_parseRPMSpec.hs]: ok, sounds like this is the expected behavior but we can't test for it b/c hspec-attoparsec doesn't allow the leftover to be Nothing. I will delete this assertion.

**Comment by atodorov** [comment]
@clumens @dashea this PR should be ready to go. I will merge tomorrow if no objections.

**Pull request merged**